flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/DynamicKafkaRecordSerializationSchemaTest.java

```diff
@@ -0,0 +1,230 @@
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** Test for DynamicKafkaRecordSerializationSchema. */
+public class DynamicKafkaRecordSerializationSchemaTest {
+
+    @Test
+    public void testserializewithoutMethod() {
+
+        DynamicKafkaRecordSerializationSchema.nullifySerializeWithAdditionalPropertiesMethod();
+        try {
+            FlinkKafkaPartitioner<RowData> partitioner = null;
+            MockSerializationSchema<RowData> keySerialization = null;
+
+            RowData.FieldGetter[] keyFieldGetters = null;
+
+            keySerialization = new MockSerializationSchema<>(true);
+            RowData.FieldGetter keyFieldGetter = RowData.createFieldGetter(new IntType(), 0);
+            keyFieldGetters = new RowData.FieldGetter[] {keyFieldGetter};
+
+            MockSerializationSchema<RowData> valueSerialization =
+                    new MockSerializationSchema<>(false);
+            RowData.FieldGetter valueFieldGetter = RowData.createFieldGetter(new IntType(), 0);
+            RowData.FieldGetter[] valueFieldGetters = {valueFieldGetter};
+
+            DynamicKafkaRecordSerializationSchema dynamicKafkaRecordSerializationSchema =
+                    new DynamicKafkaRecordSerializationSchema(
+                            "test",
+                            partitioner,
+                            keySerialization,
+                            valueSerialization,
+                            keyFieldGetters,
+                            valueFieldGetters,
+                            false,
+                            new int[] {},
+                            false);
+
+            RowData consumedRow = new GenericRowData(1);
+            KafkaRecordSerializationSchema.KafkaSinkContext context =
+                    new KafkaRecordSerializationSchema.KafkaSinkContext() {
+                        @Override
+                        public int getParallelInstanceId() {
+                            return 0;
+                        }
+
+                        @Override
+                        public int getNumberOfParallelInstances() {
+                            return 0;
+                        }
+
+                        @Override
+                        public int[] getPartitionsForTopic(String topic) {
+                            return new int[0];
+                        }
+                    };
+
+            Long timestamp = null;
+            final ProducerRecord<byte[], byte[]> producerRecord =
+                    dynamicKafkaRecordSerializationSchema.serialize(
+                            consumedRow, context, timestamp);
+            Headers headers = producerRecord.headers();
+            assertThat(headers).isEmpty();
+        } finally {
+            // ensure the method is present after this test.
+            DynamicKafkaRecordSerializationSchema.initializeMethod();
```

Comment:
Similar, I don't think it's good practice to use `finally` as a `beforeEach/afterEach`